packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -90,7 +90,22 @@ export default class Proxy {
   async destroyProxy(id) {
     const { vmUuid } = await this._getProxy(id)
     if (vmUuid !== undefined) {
-      await this._app.getXapi(vmUuid).deleteVm(vmUuid)
+      let xapi
+      try {
+        xapi = this._app.getXapi(vmUuid)
+      } catch (error) {
+        if (!noSuchObject.is(error)) {
+          throw error
+        }
+      }
+
+      await xapi?.deleteVm(vmUuid)
```

Comment:
Why not do this in the `try` block?

Reply:
It was just for the logic separation because the catch isn't for the VM deletion.

Reply:
I understand the intention, I probably would have had the same one, but I think it complicates the code with no good reasons by trying to work around native statement.

What do you think?